subsys/disk/disk_access_usdhc.c

```diff
@@ -0,0 +1,2879 @@
+	struct usdhc_data *data = &priv->op_context.data;
+
+	if (!priv->host_ready) {
+		return -ENODEV;
+	}
+
+	/* reset variables */
+	priv->card_info.card_flags = 0U;
+	/* set DATA bus width 1bit at beginning*/
+	usdhc_set_bus_width(base, USDHC_DATA_BUS_WIDTH_1BIT);
+	/*set card freq to 400KHZ at begging*/
+	priv->card_info.busclk_hz =
+		usdhc_set_sd_clk(base, priv->host_config.src_clk_hz,
+			SDMMC_CLOCK_400KHZ);
+	/* send card active */
+	ret = usdhc_set_sd_active(base);
+	if (ret == false) {
+		return -EIO;
+	}
+
+	/* Get host capability. */
+	usdhc_get_host_capability(base, &priv->host_capability);
+
+	/* card go idle */
+	usdhc_op_ctx_init(priv, 1, SDHC_GO_IDLE_STATE, 0, SDHC_RSP_TYPE_NONE);
+
+	ret = usdhc_xfer(priv);
+	if (ret) {
+		return ret;
+	}
+
+	if (USDHC_SUPPORT_V330_FLAG != SDMMCHOST_NOT_SUPPORT) {
+		app_cmd_41_arg |= (SD_OCR_VDD32_33FLAG | SD_OCR_VDD33_34FLAG);
+		priv->card_info.voltage = SD_VOL_3_3_V;
+	} else if (USDHC_SUPPORT_V300_FLAG != SDMMCHOST_NOT_SUPPORT) {
+		app_cmd_41_arg |= SD_OCR_VDD29_30FLAG;
+		priv->card_info.voltage = SD_VOL_3_3_V;
+	}
+
+	/* allow user select the work voltage, if not select,
+	 * sdmmc will handle it automatically
+	 */
+	if (USDHC_SUPPORT_V180_FLAG != SDMMCHOST_NOT_SUPPORT) {
+		app_cmd_41_arg |= SD_OCR_SWITCH_18_REQ_FLAG;
+	}
+
+	/* Check card's supported interface condition. */
+	usdhc_op_ctx_init(priv, 1, SDHC_SEND_IF_COND,
+		SDHC_VHS_3V3 | SDHC_CHECK, SDHC_RSP_TYPE_R7);
+
+	retry = 10;
+	while (retry) {
+		ret = usdhc_xfer(priv);
+		if (!ret) {
+			if ((cmd->response[0U] & 0xFFU) != SDHC_CHECK) {
+				ret = -ENOTSUP;
+			} else {
+				break;
+			}
+		}
+		retry--;
+	}
+
+	if (!ret) {
+		/* SDHC or SDXC card */
+		app_cmd_41_arg |= SD_OCR_HOST_CAP_FLAG;
+		priv->card_info.card_flags |= USDHC_SDHC_FLAG;
+	} else {
+		/* SDSC card */
+		LOG_ERR("USDHC SDSC not implemented yet!\r\n");
+		return -ENOTSUP;
+	}
+
+	/* Set card interface condition according to SDHC capability and
+	 * card's supported interface condition.
+	 */
+APP_SEND_OP_COND_AGAIN:
+	usdhc_op_ctx_init(priv, 1, 0, 0, SDHC_RSP_TYPE_NONE);
+	ret = usdhc_app_host_cmd(priv, NXP_SDMMC_MAX_VOLTAGE_RETRIES, 0,
+		SDHC_APP_SEND_OP_COND, app_cmd_41_arg,
+		SDHC_RSP_TYPE_R1, SDHC_RSP_TYPE_R3, 1);
+	if (ret) {
+		LOG_ERR("APP Condition CMD failed:%d\r\n", ret);
+		return ret;
+	}
+	if (cmd->response[0U] & SD_OCR_PWR_BUSY_FLAG) {
+		/* high capacity check */
+		if (cmd->response[0U] & SD_OCR_CARD_CAP_FLAG) {
+			priv->card_info.card_flags |= SDHC_HIGH_CAPACITY_FLAG;
+		}
+		/* 1.8V support */
+		if (cmd->response[0U] & SD_OCR_SWITCH_18_ACCEPT_FLAG) {
+			priv->card_info.card_flags |= SDHC_1800MV_FLAG;
+		}
+		priv->card_info.raw_ocr = cmd->response[0U];
+	} else {
+		goto APP_SEND_OP_COND_AGAIN;
+	}
+
+	/* check if card support 1.8V */
+	if ((priv->card_info.card_flags & USDHC_VOL_1_8V_FLAG)) {
+		usdhc_op_ctx_init(priv, 1, SDHC_VOL_SWITCH,
+			0, SDHC_RSP_TYPE_R1);
+
+		ret = usdhc_xfer(priv);
+		if (!ret) {
+			ret = usdhc_
```

Comment:
if you use generic pin definitons in dts, you can drop nxp specific definitions, same for slot 0 above